ChronosTankDrive/src/main/java/frc/robot/subsystems/shooter/ShooterSupersystem.java

```diff
@@ -13,30 +13,58 @@
     private Hood hood;
     private Limelight limelight;
     private DifferentialDrive driveTrain;
-    private GenericHID joystick;
-
+    private GenericHID joystick1,joystick2;
+    public Hood getHood(){
+        return hood;
+    }
     public ShooterSupersystem(Shooter shooter, Hood hood, Limelight limelight, DifferentialDrive driveTrain,
-            GenericHID joystick) {
+            GenericHID joystick1,GenericHID joystick2) {
         this.shooter = shooter;
         this.hood = hood;
         this.limelight = limelight;
         this.driveTrain = driveTrain;
-        this.joystick = joystick;
+        this.joystick1 = joystick1;
+        this.joystick2=joystick2;
     }
 
     @Override
     public void init() {
         shooter.init();
-        hood.init();
+        //hood.init();
         SmartDashboard.putNumber("RPM", 0);
     }
-
-    // TODO: Review this method carefully. There was a merge conflict between code
-    // written on 3/15 and on 3/16 and I
-    // tried to merge them together to maintain both contributions. -CRS
     public double hoodAngle(double distance){
         return 90-Math.atan((Constants.Shooter.SHOOTER_HEIGHT_DIFF+Math.sqrt(Math.pow(Constants.Shooter.SHOOTER_HEIGHT_DIFF,2)+Math.pow(distance,2)))/distance)*180/Math.PI;
     }
+    public boolean alignTo(double angle,double distance){
+        double turn = -angle / 25;
+        SmartDashboard.putNumber("Turn", turn);
+        if (Math.abs(turn) < .03) {
+            turn = 0;
+        }
+        driveTrain.arcadeDrive(0, turn);
+        double hoodAngle=hoodAngle(distance);
+        SmartDashboard.putNumber("Hood Angle", hoodAngle);
+        hood.setAngle(hoodAngle);
+        return turn<.2&&Math.abs(hood.getAngle()-hoodAngle)<2;
+    }
+    public boolean align(){
+        double[] distanceAndAngle = limelight.getAdjustedDistanceAndAngleToTarget();
+        double distance = distanceAndAngle[0];
+        double angle=distanceAndAngle[1];
+        return alignTo(angle, distance);
+    }
+    public void shoot(double distance){
+        shooter.shoot(4.045*distance+2374.7);
```

Comment:
@TonyBaston - with the changes we've made to the system, we should re-run the calibration experiment to make sure this line is still the best one to use.

Reply:
Do we have time for that?

Reply:
Probably not before the competition, I made the call moment to remind us that we’ve made changes that may affect the calibration, so we will want to recalibrate when we have some time.